### PR TITLE
Implemented models

### PR DESCRIPTION
**Pull request by Dunaysky**
Implemented models with specs and factories. Added simplecov in project.

### CODE REVIEW

**Review by danyil-cybercraft — COMMENTED** [review]
Please add `covarage` folder to `.gitignore` file

File: db/migrate/20230210201108_create_event_users.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreateEventUsers < ActiveRecord::Migration[7.0]
+  def change
+    create_table :event_users do |t|
+      t.integer :event_id
```
Comment (danyil-cybercraft) [inline db/migrate/20230210201108_create_event_users.rb]: You can use `t.reference` operator here

File: rvm.sh [hunk 2]
```diff
@@ -0,0 +1,939 @@
+#!/usr/bin/env bash
```
Comment (danyil-cybercraft) [inline rvm.sh]: If you need this file on your local machine, please add it to `.gitignore` file. If you do not need this file, please remove it

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+RSpec.describe User do
+  context 'when have many event_users' do
+    it { is_expected.to have_many(:event_users) }
+  end
+
+  context 'when has a valid factory' do
+    it { expect(create(:user)).to be_valid }
+  end
+
+  context 'when validate presence' do
```
Comment (danyil-cybercraft) [inline spec/models/user_spec.rb:14]: Please add tests for uniqueness and email validation

File: .rubocop.yml [hunk 4]
```diff
@@ -57,3 +57,12 @@ RSpec/AnyInstance:
   Enabled: false
 RSpec/FactoryBot/SyntaxMethods:
   Enabled: false
+# RSpec/ContextWording:
```
Comment (danyil-cybercraft) [inline .rubocop.yml]: Please uncommend this parameter, all contexts should starts with these prefixes

File: db/migrate/20230210201108_create_event_users.rb (hunk 1 above)
Comment (Dunaysky) [reply db/migrate/20230210201108_create_event_users.rb]: Used `t.references`

File: rvm.sh (hunk 2 above)
Comment (Dunaysky) [reply rvm.sh]: Removed the file

File: spec/models/user_spec.rb (hunk 3 above)
Comment (Dunaysky) [reply spec/models/user_spec.rb:14]: Added tests for uniqueness and email validation

File: .rubocop.yml (hunk 4 above)
Comment (Dunaysky) [reply .rubocop.yml]: Uncommented parameter ContextWording

File: spec/factories/event_users.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :event_user do
+    event_id { Faker::Number.within(range: 1..10) }
```
Comment (danyil-cybercraft) [inline spec/factories/event_users.rb]: You should create event and attach it to event_user
You can do it very simple
```suggestion
    event
```

File: spec/models/event_user_spec.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+  context 'when belong to user and event' do
+    it { is_expected.to belong_to(:user) }
+    it { is_expected.to belong_to(:event) }
+  end
+
+  context 'when has a valid factory' do
+    it {
+      expect(create(:event_user,
+                    user_id: user.id,
```
Comment (danyil-cybercraft) [inline spec/models/event_user_spec.rb]: After you complete [this](https://github.com/Dunaysky/event-manager/pull/2/files#r1105161756) comment. You can create even_user without passing event and user variables

File: spec/models/user_spec.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User do
+  let(:invalid_email) do
+    described_class.create(email: 'invalid',
```
Comment (danyil-cybercraft) [inline spec/models/user_spec.rb]: You can use
```suggestion
    build(:user, email: 'invalid',
```

File: spec/factories/event_users.rb (hunk 5 above)
Comment (Dunaysky) [reply spec/factories/event_users.rb]: Created `event` and `user` and attached it to `event_user` factory

File: spec/models/event_user_spec.rb (hunk 6 above)
Comment (Dunaysky) [reply spec/models/event_user_spec.rb]: Simplyfied creation of valid `event_user`: 
`it { expect(create(:event_user)).to be_valid }`

File: spec/models/user_spec.rb (hunk 7 above)
Comment (Dunaysky) [reply spec/models/user_spec.rb]: Simplified creation user with invalid email:
`it { expect(build(:user, email: 'invalid')).not_to be_valid }`

**Review by danyil-cybercraft — APPROVED** [review]

**Pull request merged**